types/d3-format/index.d.ts

```diff
@@ -44,20 +44,22 @@ export interface FormatLocaleObject {
      * Returns a new format function for the given string specifier. The returned function
      * takes a number as the only argument, and returns a string representing the formatted number.
      *
-     * @param specifier A Specifier string
+     * @param specifier A Specifier string.
+     * @throws Error on invalid format specifier.
      */
-    format(specifier: string): (n: number) => string;
+    format(specifier: string): (n: { valueOf(): number; } | string) => string;
 
```

Comment:
The matter of specifier type "c" is interesting in that the official API doc does not reflect the implemented code since [this change to the d3-format code](https://github.com/d3/d3-format/commit/e247671b89d78fa27106bafede4d7703c5ac1eb0). The official API doc still refers to the old behaviour:
```
c - converts the integer to the corresponding unicode character before printing.
```
and the reference to the argument of the returned formatter function being a "number". So good on you for finding a rather undocumented behaviour in the source.

Seems to me on yet further consideration, that (a) the API doc should be updated to reflect the current behaviour (PR to d3-format) AND (b) we need a more complex explanation of the permissible inputs to the formatter functions as well, i.e. when an actual text string is permissible in combination with format-specifiers of type "c".

So, I stand corrected on the extension to also include `string` explicitly. No-one seems to have used it so far in actual applications with these TS definitions, otherwise we would have likely heard about it.